### DISCUSSION TITLE
Sequel::DatabaseDisconnectError: PG::ConnectionBad: PQconsumeInput() SSL error: decryption failed or bad record mac

### DISCUSSION THREAD

**Discussion by zw963** [jeremyevans/sequel#1836] [General]
Hi, hello, i meet one issue about pg one month ago since i deploy my pg 14.1 (Debian 14.1-1.pgdg90+1)  on debian 10, i use postgresql official apt source repo.

it works, but i  meet following issue frequently when retrieve data.

PQconsumeInput() SSL error: decryption failed or bad record mac

The real wired things is,  i migrate my project from another VPS which use CentOS 8,  and in that server, i use pg 14.0 which install use official source repo too, but, this issue never happen.

So, i am so curious if this issue only happen on Debian? or only happen on 14.1? (not happen on 14).

I consider this should not be a sequel issue, but, could you please give some clue how to fix that?

I don't want just reinstall my VPS OS to centos or Ubuntu to workaround this. (may work, may not work anyway)

---------

Following is my pg_hba.conf

```
Following is my pg pg_hba.conf, i only use pg from 127.0.0.1

/etc/postgresql/14/main$ sudo cat pg_hba.conf  |grep -v '^#'

local   all             postgres                                peer
local   all             all                                     peer
host    all             all             127.0.0.1/32            scram-sha-256
host    all             all             ::1/128                 scram-sha-256
local   replication     all                                     peer
host    replication     all             127.0.0.1/32            scram-sha-256
host    replication     all             ::1/128                 scram-sha-256
```

Thank you!

**Comment by jeremyevans** [comment]
You are correct, this is not a Sequel issue.  Not sure what is going wrong in your environment or how to fix it, sorry.

**Comment by janko** [comment]
I get these errors often on our Azure Postgres database. Not sure what to do about them either, I just classified them as transient errors that are inevitable when using a hosted service. Maybe there are ways to get rid of it 🤷🏻‍♂️

**Reply by kandalf** [reply]
I'm facing the same for Postgres on RDS, for a small Cuba API. It's driving me crazy.
It fails immediately, it's not a timeout, and no connection option seems to make a difference.

What's weird is that I'm running a parallel Rails application, connecting to another very similar Postgres instance in the same RDS account, zone, VPC, etc, and ActiveRecord doesn't raise these errors.

If it's not a Sequel error, could it be a `pg` library error? I'm running out of ideas.
Any tips for debugging these errors? they are also random, of course.
